### PR TITLE
Fix #2563: Unconditionally emit mixin forwarders

### LINKED ISSUE #2563: Emit mixin forwarders by default?
This is what Scala 2.12 does, for startup performance reasons (http://www.scala-lang.org/blog/2016/07/08/trait-method-performance.html). Currently we're only doing it for JUnit-annotated methods (https://github.com/lampepfl/dotty/pull/2557).
/cc @DarkDimius

### PR DESCRIPTION
**Pull request by smarter**
This PR is a draft because it depends on https://github.com/lampepfl/dotty/pull/6079, only the last three commits are new.

### CODE REVIEW

**Comment by smarter** [comment]
test performance please

**Comment by liufengyun** [comment]
test performance please

**Comment by smarter** [comment]
test performance please

File: compiler/src/dotty/tools/dotc/config/ScalaSettings.scala [hunk 1]
```diff
@@ -76,6 +77,18 @@ class ScalaSettings extends Settings.SettingGroup {
...
 
+  val XmixinForceForwarders = ChoiceSetting(
+    name    = "-Xmixin-force-forwarders",
+    helpArg = "mode",
+    descr   = "Generate forwarder methods in classes inhering concrete methods from traits.",
+    choices = List("true", "junit", "false"),
+    default = "junit")
+
+  object mixinForwarderChoices {
+    def isTruthy(implicit ctx: Context) = XmixinForceForwarders.value == "true"
```
Comment (odersky) [inline compiler/src/dotty/tools/dotc/config/ScalaSettings.scala]: isTruthy? What name is that? Use `isTrue` instead?

Comment (smarter) [reply compiler/src/dotty/tools/dotc/config/ScalaSettings.scala]: This is copy-pasted from scalac, but I'm fine with changing it.

Comment (odersky) [reply compiler/src/dotty/tools/dotc/config/ScalaSettings.scala]: OK, whatever...

**Pull request merged**